### PR TITLE
Added an Faq on api versioning

### PR DESCRIPTION
**Pull request by Darshan174**
## Description

Added an Faq on api versioning.

Fixes https://github.com/istio/istio.io/issues/15319

## Reviewers

- [ ] Ambient
- [X] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [X] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

**Comment by kfaseela** [comment]
Hello @Darshan174 please sign the cla

**Comment by Darshan174** [comment]
> Hello @Darshan174 please sign the cla

Hey @kfaseela! I've signed the cla

**Comment by kfaseela** [comment]
@howardjohn let me know if this is good to go

File: content/en/about/faq/traffic-management/api.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Which API version should I use?
+weight: 10
+---
+
+Always use the primary/latest API version declared in the CRD, as this will give you the most up-to-date functionality and the system will handle any necessary version conversions transparently.
```
Comment (howardjohn) [inline content/en/about/faq/traffic-management/api.md:6]: This is not quite true, any version gives up-to-date functionality. 

I think we want to convey a message that it _it really doesn't matter_. There is no reason not to use the latest (unless you are an integration author, then see below), but there is also no strong need to spending time migrating existing resources or anything like that

Comment (Darshan174) [reply content/en/about/faq/traffic-management/api.md:6]: Hey @howardjohn, is this [link](https://istio.io/latest/docs/ops/integrations/) sufficient for displaying info on integration author in the FAQ

**Comment by craigbox** [comment]
@Darshan174 Thanks for contributing.  Docs are sometimes an easy place to start, but you really have to understand a system to write something in the language a new user to it will be able to comprehend.

I don't think what we have is going to help someone understand the problem, and I think that it is because you, as the author, don't understand the problem.

I would recommend you consider getting started with code contributions; perhaps try looking at the `istio/istio` repo for good first issues there.

**Pull request closed without merging**